libminifi/src/capi/api.cpp

```diff
@@ -287,6 +288,24 @@ flow *create_getfile(nifi_instance *instance, flow *parent_flow, GetFileConfig *
   return new_flow;
 }
 
+processor *add_processor(flow *flow, const char *processor_name) {
+  ExecutionPlan *plan = static_cast<ExecutionPlan*>(flow->plan);
+  auto proc = plan->addProcessor(processor_name, processor_name);
+  if (proc) {
+    processor *new_processor = new processor();
+    new_processor->processor_ptr = proc.get();
+    return new_processor;
+  }
+  return nullptr;
+}
+int set_property(processor *proc, const char *name, const char *value) {
+  if (name != nullptr && value != nullptr && proc != nullptr) {
+    core::Processor *p = static_cast<core::Processor*>(proc->processor_ptr);
+    return p->setProperty(name, value) ? 0 : -1;
```

Comment:
can we return a different response code here? maybe -2? If we return anything here other than zero, probably should differentiate from invalid arguments.